ext/pcntl/pcntl.c

```diff
@@ -1622,6 +1626,43 @@ PHP_FUNCTION(pcntl_getcpu)
 }
 #endif
 
+#if defined(HAVE_PTHREAD_SET_QOS_CLASS_SELF_NP)
+PHP_FUNCTION(pcntl_getqos_class)
+{
+	qos_class_t qos_class;
+
+	ZEND_PARSE_PARAMETERS_NONE();
+
+	if (UNEXPECTED(pthread_get_qos_class_np(pthread_self(), &qos_class, NULL) != 0))
+	{
+		// unlikely unless an external tool set the QOS class with a wrong value
+		PCNTL_G(last_error) = errno;
+		zend_value_error("invalid QOS class %u", qos_class);
+		RETURN_THROWS();
+	}
```

Comment:
How can this happen? Also this doesn't look like a ValueError as no user input is taken.

Reply:
true